tests/unittest/benchmark_runner/common/oc/test_singleton_login.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+from unittest.mock import MagicMock, patch
+from benchmark_runner.common.oc.oc_exceptions import LoginFailed
+from benchmark_runner.common.oc.singleton_login import SingletonLogin  # Adjust import as necessary
+
+
+@pytest.fixture
+def mock_oc_instance():
+    """Fixture to create a mocked OC instance."""
+    oc_instance = MagicMock()
+    oc_instance._kubeconfig_path = 'path/to/kubeconfig.yaml'
+    oc_instance._kubeadmin_password = 'valid_password'
+    oc_instance._cli = 'oc'
+    oc_instance.RETRIES = 3
+    oc_instance.DELAY = 1
+    return oc_instance
+
+
```

Comment:
Create the TestSingletonLogin (or TestSingletonOCLogin) class here.  Maybe ResetSingletonOCLogin would be an even better name for the class since it's only being used for that purpose; everything else just uses SingletonLogin/SingletonOCLogin.

Reply:
I have changed the file name to test_singleton_oc_login.py.
No need to create a class TestSingletonOCLogin, I am using pytest and not unittest.

Reply:
It's to avoid having that dangerous reset member available.  It's only needed for testing and should not be accessible outside of testing.

TestSingletonOCLogin is not correct, as pytest keys off the name.  Perhaps ResetSingletonOCLogin would be a better name for it.

Reply:
I add the following fixture to pytest

```
@pytest.fixture(autouse=True)
def reset_singleton():
    """Fixture to reset SingletonOCLogin before each test."""
    with SingletonOCLogin._lock:
        SingletonOCLogin._instance = None
```
and delete reset_instance method from SingletonOCLogin class
```

    @classmethod
    def reset_instance(cls):
        with cls._lock:  # Acquire the lock for instance reset
            cls._instance = None
        
```